core/src/main/scala/kafka/cluster/Broker.scala

```diff
@@ -88,7 +90,7 @@ object Broker {
                 (ep.protocolType, ep)
               }.toMap
             }
-          val rack: Option[String] = brokerInfo.get("rack").asInstanceOf[Option[String]]
+          val rack = brokerInfo.get("rack").filter(_ != null).map(_.asInstanceOf[String])
```

Comment:
OK, I digged a little more on this. The code I wrote worked because I never include null rack in JSON string when the broker registers with ZK. That's why the last test in `AdminRackAwareTest` passes which confused me a little bit when I found `rack.nonEmpty` does not return the expected result.

I think your code will work even if null rack is leaked to ZK. So I am fine with your code.